### ISSUE TITLE
Invalid YAML being generated for certain inputs

### ISSUE THREAD

**Issue by jan-tee** [eemeli/yaml#237]
**Describe the bug**

When certain input is encountered, the YAML output is invalid. When trying to parse it, an exception is thrown.

**To Reproduce**

Repro repository is here: https://github.com/jan-tee/yaml-bug-1
Includes test for the same case with `js-yaml` without issues.

**Expected behaviour**

YAML output is valid, can be parsed without exception.

**Versions (please complete the following information):**

- Environment: Node v15.8.0, Windows
- `yaml`: 1.10.0

YAML output looks like:

```
test:
  test:
    test:
      - test: "\r

          \\XXXXXX\\XXXXXXXXXXXXXXX\\XXXXXXXXXXX\\XXXXX\\XXXXX\\XXXX\\
          \X\\XXXXX\\XXXXX\\XXXXXXXXX\\XXXXXXXXX\\XXXXXXXXX\\XXX_XXXXXX_X.X_(XX\
          XXXXXXXX"
```

Exception when parsing the YAML back:

```
YAMLSyntaxError: Invalid escape sequence \X
    at QuoteDouble.get strValue [as strValue] (C:\Users\jant.SENTINELONE\Repos\jan-tee\yaml-bug-1\node_modules\yaml\dist\parse-cst.js:1327:25)
    at Object.resolveString [as resolve] (C:\Users\jant.SENTINELONE\Repos\jan-tee\yaml-bug-1\node_modules\yaml\dist\resolveSeq-4a68b39b.js:1282:20)
    at resolveByTagName (C:\Users\jant.SENTINELONE\Repos\jan-tee\yaml-bug-1\node_modules\yaml\dist\resolveSeq-4a68b39b.js:1384:25)
    at resolveTag (C:\Users\jant.SENTINELONE\Repos\jan-tee\yaml-bug-1\node_modules\yaml\dist\resolveSeq-4a68b39b.js:1414:17)
    at resolveNodeValue (C:\Users\jant.SENTINELONE\Repos\jan-tee\yaml-bug-1\node_modules\yaml\dist\resolveSeq-4a68b39b.js:1535:23)
    at resolveNode (C:\Users\jant.SENTINELONE\Repos\jan-tee\yaml-bug-1\node_modules\yaml\dist\resolveSeq-4a68b39b.js:1583:15)
    at resolveBlockMapItems (C:\Users\jant.SENTINELONE\Repos\jan-tee\yaml-bug-1\node_modules\yaml\dist\resolveSeq-4a68b39b.js:1777:38)
    at Object.resolveMap [as resolve] (C:\Users\jant.SENTINELONE\Repos\jan-tee\yaml-bug-1\node_modules\yaml\dist\resolveSeq-4a68b39b.js:1612:79)
    at resolveByTagName (C:\Users\jant.SENTINELONE\Repos\jan-tee\yaml-bug-1\node_modules\yaml\dist\resolveSeq-4a68b39b.js:1384:25)
    at resolveTag (C:\Users\jant.SENTINELONE\Repos\jan-tee\yaml-bug-1\node_modules\yaml\dist\resolveSeq-4a68b39b.js:1414:17) {
  source: QuoteDouble {
    error: null,
    range: Range { start: 38, end: 214 },
    valueRange: Range { start: 38, end: 214 },
    props: [],
    type: 'QUOTE_DOUBLE',
    value: null,
    resolved: Scalar {
      value: '\r\n' +
        '\\XXXXXX\\XXXXXXXXXXXXXXX\\XXXXXXXXXXX\\XXXXX\\XXXXX\\XXXX\\ \\X\\XXXXX\\XXXXX\\XXXXXXXXX\\XXXXXXXXX\\XXXXXXXXX\\XXX_XXXXXX_X.X_(XXXXXXXXXX',
      range: [ 38, 214 ],
      type: 'QUOTE_DOUBLE'
    }
  }
}
```

**Comment by eemeli** [comment]
This looks like a duplicate of awslabs/cdk8s#494, which got fixed in `yaml@2.0.0-3`. It looks like installing that version in your repo also makes the test succeed.

**Comment by eemeli** [comment]
Fixed in [`yaml@1.10.1`](https://github.com/eemeli/yaml/releases/tag/v1.10.1).
